util/setuidgid.c

```diff
@@ -213,9 +213,15 @@ string_extract_user_group(const char * combined, char ** username_p,
 	/* Sanity check. */
 	assert(combined != NULL);
 
-	/* Search for ":" and get string length. */
-	pos = strcspn(combined, ":");
+	/* Reject a 0-length string. */
 	len = strlen(combined);
+	if (len == 0) {
```

Comment:
Not sure I like this.  I would handle this case by changing the conditions below:

1. `if (pos == 0 && len > 0)` for detecting `:group`,
2. `if (pos != len)` for detecting `user:group`,
3. `if (len > 0)` for detecting `user`,

and put a

```
else {
    warn0("Invalid user:group string: \"%s\", combined);
    goto err0;
}
```

at the end.

Reply:
It amounts to the same thing but I feel like it's clearer to be specifying conditions which correspond to the different cases rather than special-casing empty strings.